test/e2e/vsphere_volume_ops_strom.go

```diff
@@ -0,0 +1,188 @@
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stype "k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/api/v1"
+	storage "k8s.io/kubernetes/pkg/apis/storage/v1beta1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	vsphere "k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+/*
+	Test to perform Disk Ops storm.
+
+	Steps
+    	1. Create storage class for thin Provisioning.
+    	2. Create 30 PVCs using above storage class in annotation, requesting 2 GB files.
+    	3. Wait until all disks are ready and all PVs and PVCs get bind. (CreateVolume strom)
+    	4. Create pod to mount volumes using PVCs created in step 2. (AttachDisk strom)
+    	5. Wait for pod status to be running.
+    	6. Verify all volumes accessible and available in the pod.
+    	7. Delete pod.
+    	8. wait until volumes gets detached. (DetachDisk strom)
+    	9. Delete all PVCs. This should delete all Disks. (DeleteVolume strom)
+	10. Delete storage class.
+*/
+
+var _ = framework.KubeDescribe("volume operations strom [Volume]", func() {
+	f := framework.NewDefaultFramework("volume-ops-strom")
+	var (
+		client            clientset.Interface
+		namespace         string
+		storageclass      *storage.StorageClass
+		pvclaims          []*v1.PersistentVolumeClaim
+		persistentvolumes []*v1.PersistentVolume
+		err               error
+		volume_ops_scale  int
+	)
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		if len(nodeList.Items) == 0 {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		if os.Getenv("VOLUME_OPS_SCALE") != "" {
+			volume_ops_scale, err = strconv.Atoi(os.Getenv("VOLUME_OPS_SCALE"))
+			Expect(err).NotTo(HaveOccurred())
+		} else {
+			volume_ops_scale = 30
+		}
+	})
+	AfterEach(func() {
+		By("Deleting PVCs")
+		for _, claim := range pvclaims {
+			framework.DeletePersistentVolumeClaim(client, claim.Name, namespace)
+		}
+		By("Deleting StorageClass")
+		err = client.StorageV1beta1().StorageClasses().Delete(storageclass.Name, nil)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should create pod with many volumes and verify none attach call fails", func() {
+		By(fmt.Sprintf("Running test with VOLUME_OPS_SCALE: %v", volume_ops_scale))
+		By("Creating Storage Class")
+		scParameters := make(map[string]string)
+		scParameters["diskformat"] = "thin"
+		storageClassSpec := getVSphereStorageClassSpec("thinsc", scParameters)
+		storageclass, err = client.StorageV1beta1().StorageClasses().Create(storageClassSpec)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating PVCs using the Storage Class")
+		count := 0
+		for count < volume_ops_scale {
+			pvclaimSpec := getVSphereClaimSpecWithStorageClassAnnotation(namespace, storageclass)
```

Comment:
tried using Spec.storageClassName on 1.5.3, but pvc is created without storage class name. since this is still being deprecated, for now using StorageClassAnnotation